service/ui.go

```diff
@@ -312,3 +322,27 @@ func (uis *UIServer) LoggedError(w http.ResponseWriter, r *http.Request, code in
 		http.Error(w, err.Error(), code)
 	}
 }
+
+func (uis *UIServer) GetCommonViewData(w http.ResponseWriter, r *http.Request, needsUser, needsProject bool) ViewData {
+	viewData := ViewData{}
+	userCtx := GetUser(r)
+	if needsUser && userCtx == nil {
```

Comment:
When should the caller set `needsUser`? (`GetUser(r)` is always called.)

Reply:
Added a comment to the function on how it's used